lib/json_schema/json_api_missing_attribute.rb

```diff
@@ -12,8 +12,10 @@ def initialize(details)
     def to_human(detail)
       if detail['type'] == 'required'
         "The property #{to_source(detail)} did not contain the required key #{detail['details']['missing_key']}"
+      elsif detail['type'] == 'schema' && detail['schema_pointer'].end_with?('additionalProperties')
```

Comment:
it was hard for me to tell looking at the validation gem if `additionalProperties` would be the only type of validation with type `schema`, but adding in the `schema_pointer` check as well, should be sufficient to ensure that we're catching the right error